sync/limiter_test.go

```diff
@@ -155,6 +162,33 @@ func TestLimiter(t *testing.T) {
 		require.EqualValues(t, 10, counterLow, "low priority counter should be 10")
 	})
 
+	t.Run("with sleep", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		var wg sync.WaitGroup
+
+		sleepTime := 1 * time.Millisecond
+		limiter := miscsync.NewLimiter(ctx, &wg, "test", 1, stats.NOP, miscsync.WithLimiterDynamicPeriod(sleepTime/100))
+		var counter int
+		var sleepVerified bool
+		for i := range 1000 {
+			wg.Add(1)
+			key := strconv.Itoa(i)
+			go func() {
+				defer wg.Done()
+				le := limiter.BeginWithSleep(key)
+				defer le.End()
+				if !sleepVerified && counter == 0 && i > 0 {
+					sleepVerified = true
+				}
+				require.NoError(t, le.Sleep(context.Background(), sleepTime))
+				counter++ // since the limiter's limit is 1, we shouldn't need an atomic counter
+			}()
+		}
+		cancel()
+		wg.Wait()
+		require.True(t, sleepVerified, "sleep should have been verified")
+	})
```

Comment:
do we want to validate some sleep related stats that we capture? like `totalSleeping` is at least > sleep duration?